pkg/metaclient/mockclient.go

```diff
@@ -1,55 +1,618 @@
 package metaclient
 
-import "context"
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
 
-type mockKVClient struct {
-	endpoint string
-	// [TODO]
+	cerrors "github.com/hanfei1991/microcosm/pkg/errors"
+)
+
+type kvclientMock struct {
+	sync.Mutex
+
+	endpoint  string
+	clusterID string
+
+	// all expectations in mock
+	expects []expectation
+	// current expectation pos
+	fulfilled int
 }
 
-func NewMockKVClient(endpoint string) KVClient {
-	return &mockKVClient{endpoint}
+// default size for expectation internal implement
+const (
+	DefaultExpectationSize = 10
+	DefaultOpSize          = 10
+	DefaultResultSize      = 10
+)
+
+// NewMockKVClient create a kvclient and kvclient mock
+func NewMockKVClient(endpoint string, clusterID string) (KVClient, KVClientMock) {
```

Comment:
Is there any plan to replace `MetaMock` with this new mock. https://github.com/hanfei1991/microcosm/blob/836a23aa66005f1bd9fe6a4ed6a062deb8dcc3ff/pkg/metadata/kv_mock.go#L47